share/functions/help.fish

```diff
@@ -23,7 +23,7 @@ function help --description 'Show help for the fish shell'
 	#
 	set -l graphical_browsers htmlview x-www-browser firefox galeon mozilla konqueror epiphany opera netscape rekonq google-chrome chromium-browser
 
-	if set -q fish_help_browser[1]
+	if set -q fish_help_browser
```

Comment:
@faho lead you astray. The two forms are not equivalent. The `if set -q fish_help_browser[1]` is safer as it ensure the var has at least one element. The form without the subscript will evaluate to true if the user did `set fish_help_browser`; i.e., defined it without giving it a value. That's an odd thing to do but it's good practice to protect against it.